### ISSUE TITLE
Updated SwerveModule and SwerveModuleConfig

### ISSUE THREAD

**Issue by SoZ0** [FRC-6390/Crecendo-2024#2]
# The problem

Currently the SwerveModule and SwerveModuleConfig do not contain all the information related to their systems and shuld track all related date to their respective roles. They also suffer from minor bloat and dependency issues.

## Vision

The `SwerveModule` should get **ALL** the required information they need from the constructor. All calls to the constants should **REMOVED** so that the modules can be used in a more generic way.

The `SwerveModuleConfig` is missing some of the information it should contain, specfically the location of the module on the robot, gear ratio and encoder ticks per rotation.

All records including a new `SwerveModuleConstraints` will be placed inside the same file

File: ```SwerveModule.java```

Location: ```src/main/java/frc/robot/utils/drivetrain/swerve```

---

# Variables

Remove all unused variables from the `SwerveModule` and `SwerveModuleConfig`.
add the new required tracked values by keeping an instance of the `SwerveModuleConfig` class in the `SwerveModule`.
make a new record class in the `SwerveModuleConstraints` class to track all the information related to the constraints like max speeds.

## `public ShuffleboardLayout shuffleboard()`
Will create the shuffleboard layout contain all relevant data

## Tasks

- [x] **SwerveModule refactor**
- [x] **SwerveModuleConfig refactor**
- [ ] **SwerveModuleConstraints creation**
- [ ] **Shuffleboard**

**Comment by MahditS** [comment]
I dont think we should have a seperate record for swerve module contraints, as the only constraint called in swerve module is MaxSpeed MPS, and I just put it into SwerveModule Config. Also, swerve module locations should be in the swerve module config? I think that should be apart of the drivetrain generalization issue #3.

**Comment by MahditS** [comment]
Also mathias, for the telemetry, i was thinking we should have a seperate subsystem that handles all the telemetry for every single command and subsystem. Like a telemetry manager class that updates the shuffleboard in periodic. Idk thats just me tho. i would like to have all the shuffleboard stuff in one place.

**Comment by SoZ0** [comment]
> Also mathias, for the telemetry, i was thinking we should have a seperate subsystem that handles all the telemetry for every single command and subsystem. Like a telemetry manager class that updates the shuffleboard in periodic. Idk thats just me tho. i would like to have all the shuffleboard stuff in one place.

That is the idea for the telemetry issue, the shuffleboard method in the other issues is to supply that issue

**Comment by SoZ0** [comment]
> I dont think we should have a seperate record for swerve module contraints, as the only constraint called in swerve module is MaxSpeed MPS, and I just put it into SwerveModule Config. 

There is more that one constraint for the modules, the max angular as well, the idea of having it being its own record would allow you to make one object in constants and pass the same value into each module config without need to update multiple values


> Also, swerve module locations should be in the swerve module config? I think that should be apart of the drivetrain generalization issue #3.

The module location being apart of the swerve module keeps all information related to each module within the same class, keeps the data together.

**Comment by SoZ0** [comment]
Also please do not close the issue until the code have been reviewed. Best we come to a solution we all agree on first (or as close as we can get)
